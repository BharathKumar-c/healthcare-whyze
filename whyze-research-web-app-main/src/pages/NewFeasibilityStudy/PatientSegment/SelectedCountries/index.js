import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row, Switch } from 'antd';
import { Location } from '../../../../assets';
import { Text } from '../../../../components/basic';

export default function SelectedCountries({ countriesList, isShowSwitch }) {
  return (
    <div className="selected-countries-container">
      <Text className="selected-countries-title" text="Selected countries" />
      {countriesList.map((country, index) => (
        <Row
          key={index}
          className={
            isShowSwitch
              ? 'selected-countries-component-border'
              : 'selected-countries-component'
          }
        >
          <Col span={isShowSwitch ? 24 : 15}>
            <Row>
              <Col span={4}>
                <Location className="selected-countries-location-icon" />
              </Col>
              <Col className="selected-countries-text" span={20}>
                {country?.country}
              </Col>
            </Row>
          </Col>
          <Col span={9}>
            {!isShowSwitch && (
              <Row>
                <Col className="selected-countries-no-of-patients" span={15}>
                  {country?.count}
                </Col>
                <Col span={9}>
                  <Switch size="small" />
                </Col>
              </Row>
            )}
          </Col>
        </Row>
      ))}
    </div>
  );
}

SelectedCountries.propTypes = {
  countriesList: PropTypes.array,
  isShowSwitch: PropTypes.bool,
};

SelectedCountries.defaultProps = {
  countriesList: [],
  isShowSwitch: false,
};
