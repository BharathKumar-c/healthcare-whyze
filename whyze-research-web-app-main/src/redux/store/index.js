import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
