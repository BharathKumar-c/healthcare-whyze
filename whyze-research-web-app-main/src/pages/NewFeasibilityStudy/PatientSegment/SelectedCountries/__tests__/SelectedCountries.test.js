import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../../redux/store';
import SelectedCountries from '..';
import React from 'react';

const democountryList = [
  { Country: 'United Kingdom' },
  { Country: 'Albania', count: '1' },
  { Country: 'Brazil', count: '4' },
  { Country: 'Ireland', count: '4' },
];
function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}
describe('create tests for SelectedCountry', () => {
  test('create snapshots', () => {
    const { container } = renderWithContext(<SelectedCountries />);
    expect(container).toMatchSnapshot();
  });
  test('All country render corrctly', () => {
    const { container } = renderWithContext(
      <SelectedCountries countriesList={democountryList} isShowSwitch />,
    );
    expect(container).toBeInTheDocument();
  });
});
