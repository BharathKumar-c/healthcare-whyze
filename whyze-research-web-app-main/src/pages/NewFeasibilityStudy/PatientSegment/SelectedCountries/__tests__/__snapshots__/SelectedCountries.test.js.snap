// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create tests for SelectedCountry create snapshots 1`] = `
<div>
  <div
    class="selected-countries-container"
  >
    <span
      class="ant-typography selected-countries-title"
    >
      Selected countries
    </span>
  </div>
</div>
`;
